File: Dataset/Leetcode/test/67/522.py
class Solution:
    def XXX(self, a: str, b: str) -> str:
        # 从后往前
        a = a[::-1]
        b = b[::-1]
        ans = ''
        forward = 0
        i=0
        N = len(a)
        M = len(b)

        # 模2为当前值，除2为向前给的值
        while i<N and i<M:
            cur = int(a[i])+int(b[i]) + forward
            ans += str(cur%2)
            forward = cur//2
            i += 1
        if i<N:
            while i<N:
                cur = int(a[i]) + forward
                ans += str(cur%2)
                forward = cur//2
                i += 1

        if i<M:
             while i<M:
                cur = int(b[i]) + forward
                ans += str(cur%2)
                forward = cur//2
                i += 1
        if forward!=0:
            ans += str(forward)
        return ans[::-1]

File: Dataset/Leetcode/test/38/522.java
 class Solution {
    public String XXX(int n) {
        if(n==1){
            return "1";
        }

        String [] data=new String[n];
        data[0]="1";
        data[1]="11";
        for(int i=2;i<n;i++){
            int left=0;
            StringBuilder sb=new StringBuilder();
            for(int j=1;j<data[i-1].length();j++){
                if(data[i-1].charAt(j)!=data[i-1].charAt(j-1)){
                    sb.append(j-left);
                    sb.append(data[i-1].charAt(j-1));
                    left=j;
                }
                
                if(j==data[i-1].length()-1){
                    sb.append(j-left+1);
                    sb.append(data[i-1].charAt(j));
                }
            }
            data[i]=sb.toString();
            // System.out.println(data[i]);
        }

        return data[n-1];
    }
}

